components/org.wso2.carbon.identity.media.core/src/main/java/org/wso2/carbon/identity/media/core/file/FileBasedStorageSystemImpl.java

```diff
@@ -90,11 +89,11 @@ public String addMedia(List<InputStream> inputStreams, MediaMetadata mediaMetada
         } catch (IOException e) {
             throw new StorageSystemServerException("Error while uploading media to file system.", e);
         }
-
     }
 
     @Override
-    public DataContent getFile(String id, String tenantDomain, String type) throws StorageSystemException {
+    public DataContent getFile(String id, String tenantDomain, String type) throws StorageSystemServerException,
```

Comment:
this is not needed.. you can return the base exception class and then from the API component do an instance check..